chat/src/main/java/bisq/chat/Citation.java

```diff
@@ -17,24 +17,33 @@
 
 package bisq.chat;
 
+import bisq.common.annotation.ExcludeForHash;
 import bisq.common.proto.NetworkProto;
 import bisq.common.validation.NetworkDataValidation;
 import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.ToString;
 
+import java.util.Optional;
+
 @Getter
 @ToString
 @EqualsAndHashCode
 public final class Citation implements NetworkProto {
     public static final int MAX_TEXT_LENGTH = 1000;
 
-    private final String authorUserProfileId, text, chatMessageId;
+    private final String authorUserProfileId;
+    private final String text;
+
+    // Added with v2.1.7
+    @EqualsAndHashCode.Exclude
+    @ExcludeForHash
+    private final String chatMessageId;
```

Comment:
This should be optional as well to reflect that the field can be left empty. Using empty strings or null are less explicit.